agent/bench-scripts/pbench-fio.md

```diff
@@ -0,0 +1,285 @@
+
+This page describes how to use pbench-fio to do storage performance testing.  It includes these topics:
+
+* [random](#random-workloads) and [sequential](#sequential-workloads) workload parameters
+* [cache dropping](#cache-dropping)
+* [when and how](#warming-up-the-system) to ramp up workload
+* ensure [persistent writes](#data-persistence)
+* [rate limiting](#limiting-IOPS)
+* measure [latency percentile change over time](#measuring-latency-percentiles)
+* example from [OpenStack Cinder testing](#example-of-openstack-cinder-volumes)
+
+Before we go into the [syntax details](#syntax) of pbench-fio parameters, 
+which are different in some cases than the fio input parameters, 
+we describe why pbench-fio exists, and how to make use of it.
+
+# overview of usage opportunities and constraints
+
+pbench-fio was created to automate sets of fio tests, 
+including calculations of statistics for throughputs and latency.  
+The fio benchmark itself does not do this.
+
+pbench-fio also was created with distributed storage performance testing in mind -- 
+it uses the **fio --client** option to allow fio to start up a storage performance test 
+on a set of hosts simultaneously and collect all the results to a central location.
+
+pbench-fio supports ingestion of these results into a repository, 
+such as elastic search, through JSON-formatted results that are easily parsed.
+Traditional fio output is very difficult to parse correctly 
+and the resulting parser is hard to understand and likely to break as fio evolves over time.
+
+pbench-fio also provides additional postprocessing for the new fio latency histogram logs.   This feature lets you graph latency percentiles cluster-wide as a function of time.  These graphs are really important when you are running very long tests and want to look at variations in latency caused by operational events such as node, disk or network failure/replacement.
```

Comment:
Perhaps add a reference to the fio's project docs on the new fio latency histogram work?